### PR TITLE
feat: builder doesn't have correct generic when initialized through class-string

### PR DESCRIPTION
**Pull request by JeRabix**
- [x] Added or updated tests

**Changes**

Changes/fixes for #2021

### CODE REVIEW

**Comment by JeRabix** [comment]
Add stub for `Illuminate\Database\Eloquent\Model` class.
`query` method exist generic annotation only from 11 version

**Comment by JeRabix** [comment]
I add Model class stub for - 9.x version too, but I doubt that this is the right decision.

File: stubs/9.0.0/Model.stub
Comment (calebdw) [inline stubs/9.0.0/Model.stub:1]: This is not needed, we already have the `stubs/common/Model.stub` and the lowest version we support is 9.x.

File: stubs/10.0.0/Model.stub
Comment (calebdw) [inline stubs/10.0.0/Model.stub:1]: There should be no need for this stub either---why was this added? It can likely just be added to the `common/Model.stub`

**Review by calebdw — CHANGES_REQUESTED** [review]

File: stubs/10.0.0/Model.stub
Comment (JeRabix) [reply stubs/10.0.0/Model.stub:1]: I proceeded from the code that exists. And in it, many stubs are duplicated. I thought the logic is that if in version 11, for example, the declaration of a method changes, then the stub should not affect the previous version. I removed the stubs for versions 9/10, modified the common stub. Sorry, this is only my second commit to this repository. 

Let's see what the tests show now, but I think everything will be ok. I checked it on version 9 locally.

Comment (calebdw) [reply stubs/10.0.0/Model.stub:1]: We only create new stubs when there's something drastically different (like a new method introduced in 11.x that need its signature overriden). Unfortunately, this means that the stub has to be mostly duplicated.

File: stubs/common/Model.stub [hunk 1]
```diff
@@ -66,6 +66,13 @@ abstract class Model implements \JsonSerializable, \ArrayAccess
      * @return \Illuminate\Database\Eloquent\Builder<static>
      */
     public static function with($relations);
+
+    /**
+     * Begin querying the model.
+     *
+     * @return \Illuminate\Database\Eloquent\Builder<static>
+     */
+    public static function query();
```
Comment (calebdw) [inline stubs/common/Model.stub]: This is fine, however, I'm 90% sure that this stub is not being used given that there's an extension that calculates what the return type is

**Review by calebdw — APPROVED** [review]
Looks good to me, thanks!

File: stubs/common/Model.stub (hunk 1 above)
Comment (JeRabix) [reply stubs/common/Model.stub]: Without this stub method - tests in 9,10 version laravel failed

Comment (calebdw) [reply stubs/common/Model.stub]: That's because in your test you have:

```php
assertType('Illuminate\Database\Eloquent\Builder<App\User>', (new $className())->query());
```

which is calling a **static** function non-statically (and thus the static extension is not being used). This should be:

```php
assertType('Illuminate\Database\Eloquent\Builder<App\User>', (new $className())->newQuery());
```

**Comment by canvural** [comment]
Thank you!
